js/interface.js

```diff
@@ -315,14 +313,100 @@
     });
   }
 
-  function loadCustomMenuWidgets() {
-    $('.menu-styles-wrapper').addClass('loading');
+  function isNewerVersion(latestVersionMenu, currentMenuVersion) {
+    for (var index = 0; index < latestVersionMenu.length; index++) {
+      if (isNaN(+latestVersionMenu[index])) latestVersionMenu[index] = 0;
+
+      if (isNaN(+currentMenuVersion[index])) currentMenuVersion[index] = 0;
+
+      if (+latestVersionMenu[index] > +currentMenuVersion[index]) return true;
+
+      if (+latestVersionMenu[index] < +currentMenuVersion[index]) return false;
+    }
+
+    return false;
+  }
+
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function getLatestMenuVersion(data) {
+    var latestVersionMenu;
+
+    data.forEach(function(menu) {
+      var formattedMenuVersion = menu.version.split('.');
+      var formattedLatestVersionMenu = latestVersionMenu ? latestVersionMenu.version.split('.') : null;
+
+      if (latestVersionMenu) {
+        latestVersionMenu = isNewerVersion(formattedLatestVersionMenu, formattedMenuVersion) ? latestVersionMenu : menu;
+      }
+
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Creates a list of menu widgets to be displayed based on versions and current usage
+   * @param {Array} menus - List of menu widgets
+   * @returns {Array} List of menu widgets to be displayed based on versions and current usage
+   */
 
+  function generateMenuList(menus) {
+    var menusByPackage = _.groupBy(menus, 'package');
+    var menuList = [];
+
+    _.forIn(menusByPackage, function(menuVersions) {
+      var currentVersion = _.find(menuVersions, function(menuVersions) {
+        return menuVersions.instances.length;
+      });
+
+      // A version is currently in use
+      if (currentVersion) {
+        return menuList.push(currentVersion);
+      }
+
+      // Find latest version of menu
+      menuList.push(getLatestMenuVersion(menuVersions));
+    });
+
+    // Sort displayed menus by display name
+    return _.sortBy(menuList, 'name');
```

Comment:
This will sort the following menus:

- AAA
- AAC
- aab

...as

1. AAA
2. AAC
3. aab

We need to ignore casing and sort it as

1. AAA
2. aab
3. AAC